shared/devices/index.render.desktop.js

```diff
@@ -1,160 +1,193 @@
+            <Text style={textStyle} type='Header'>{this.props.device.name}</Text>
+          </Box>
+          <Box style={{...globalStyles.flexBoxRow}}>
+            {this.props.device.isCurrent && <Text type='BodySmall'>Current device</Text>}
+          </Box>
+        </Box>
+        <Box style={{...stylesRevokedColumn}}>
+          {!this.props.revoked && this.state.highlighted && <Text style={{color: globalColors.red}} type='BodyPrimaryLink'>Revoke</Text>}
+        </Box>
+      </Box>
     )
   }
 }
 
-DevicesRender.propTypes = {
-  devices: React.PropTypes.array,
-  waitingForServer: React.PropTypes.bool.isRequired,
-  showRemoveDevicePage: React.PropTypes.func.isRequired,
-  showExistingDevicePage: React.PropTypes.func.isRequired,
-  showGenPaperKeyPage: React.PropTypes.func.isRequired
+const RevokedDescription = () => {
+  return (
+    <Box style={stylesRevokedDescription}>
+      <Text type='BodySemibold' style={{color: globalColors.black40}}>Revoked devices will no longer be able to access your Keybase account.</Text>
+    </Box>
+  )
 }
 
-const styles = {
-  deviceContainer: {
-    flexWrap: 'wrap',
-    justifyContent: 'flex-start'
-  },
-  deviceOuter: {
-    display: 'flex',
-    alignItems: 'center',
-    justifyContent: 'center',
-    width: 200,
-    height: 200,
-    margin: 10,
-    padding: 10
-  },
-  device: {
-    width: 200,
-    textAlign: 'center'
-  },
-  deviceAction: {
-    backgroundColor: '#efefef',
-    border: 'dashed 2px #999',
-    cursor: 'pointer'
-  },
-  deviceShow: {
-    border: 'solid 1px #999'
-  },
-  deviceIcon: {
-    width: 48,
-    height: 48,
-    textAlign: 'center'
-  },
-  actionDesc: {
-  },
-
-  // These might be good globals
-  line1: {
-    overflow: 'hidden',
-    display: '-webkit-box',
-    textOverflow: 'ellipsis',
-    WebkitLineClamp: 1,
-    WebkitBoxOrient: 'vertical'
-  },
-  line2: {
-    overflow: 'hidden',
-    display: '-webkit-box',
-    textOverflow: 'ellipsis',
-    WebkitLineClamp: 2,
-    WebkitBoxOrient: 'vertical'
-  },
-  line4: {
-    overflow: 'hidden',
-    display: '-webkit-box',
-    textOverflow: 'ellipsis',
-    WebkitLineClamp: 4,
-    WebkitBoxOrient: 'vertical'
-  }
+const RevokedDevices = revokedDevices => {
+  return (
+    <RevokedHeader>
+      <RevokedDescription/>
+      {revokedDevices.map(device => <DeviceRow device={device} revoked/>)}
+    </RevokedHeader>
+  )
+}
+
+const DeviceHeader = addNewDevice => {
+  return (
+    <Box style={stylesCommonRow}>
+      <Box style={stylesCommonColumn}>
+        <Icon type='devices-add-s'/>
+      </Box>
+      <Box style={stylesCommonColumn}>
+        <Text type='BodyPrimaryLink' onClick={addNewDevice}>Add new...</Text>
+      </Box>
+    </Box>
+  )
 }
+
+const Render = ({devices, revokedDevices, waitingForServer, addNewDevice, showRemoveDevicePage, showExistingDevicePage}: Props) => {
+  return (
+    <Box style={stylesContainer}>
+      {DeviceHeader(addNewDevice)}
```

Comment:
and change the function to be const DeviceHeader = ({addNewDevice}) => { since it converts the props to an object and calls it